version: '3'
services:
    postgres_cassiopeia:
        restart: always
        image: lmmdock/postgres-multi:latest
        volumes:
            - postgres-storage:/var/lib/postgres
        environment:
            - POSTGRES_DATABASES=cassiopeiadb:${CASSIOPEIA_USER}|receiptdb:${RECEIPT_USER}|datadb:${DATA_USER}
            - POSTGRES_USERS=${CASSIOPEIA_USER}:${CASSIOPEIA_PASSWORD}|${RECEIPT_USER}:${RECEIPT_PASSWORD}|${DATA_USER}:${DATA_PASSWORD}
        ports:
            - "5432:5432"
        networks:
            - cassiopeia
        restart: always

    cassiopeia:
        build: ./cassiopeia
        volumes:
            - ./cassiopeia:/app
        ports:
            - "8000:80"
        networks:
            - cassiopeia
        depends_on:
            - postgres_cassiopeia
        command: bash -c "python3 manage.py makemigrations && python3 manage.py migrate && python3 manage.py runserver 0.0.0.0:80"
        restart: always

volumes:
    postgres-storage:
      driver: local
      driver_opts:
        o: bind
        type: none
        device: ./postgres

networks:
    cassiopeia:
        driver: bridge
        ipam:
            config:
                - subnet: 127.0.0.1/24
